infrasim/ipmi.py

```diff
@@ -11,23 +14,39 @@ def get_ipmi():
     except CommandRunFailed as e:
         raise CommandNotFound("/usr/local/bin/ipmi_sim")
 
+
 def status_ipmi():
     try:
         run_command("pidof ipmi_sim")
         print "InfraSim IPMI service is running"
     except CommandRunFailed as e:
         print "Infrasim IPMI service is stopped"
- 
-def start_ipmi(node):
-    ipmi_cmd = get_ipmi()
-    ipmi_start_cmd = "{0} -c /etc/infrasim/vbmc.conf" \
-                    " -f /usr/local/etc/infrasim/{1}/{1}.emu -n -s /var/tmp > /var/log/openipmi.log &".format(ipmi_cmd, node)
+
+
+def start_ipmi(conf=VM_DEFAULT_CONFIG):
     try:
-        run_command(ipmi_start_cmd, True, None, None)
-        logger.info("ipmi start")
+        with open(conf, 'r') as f_yml:
+            conf = yaml.load(f_yml)
+        if "bmc" in conf:
+            bmc = CBMC(conf["bmc"])
+        else:
+            bmc = CBMC()
+        bmc.set_type(conf["type"])
+        bmc.init()
+        bmc.precheck()
+        cmd = "{} > /var/log/openipmi.log &".format(bmc.get_commandline())
+        logger.debug(cmd)
+        run_command(cmd, True, None, None)
```

Comment:
it's better to remember the ipmisim process id for killing the process with SIGTERM not SIGKILL, since most of the application will have the signal handling,  the application could catch the SIGTERM, but the SIGKILL couldn't be caught.